usaspending_api/agency/tests/integration/test_award_count.py

```diff
@@ -0,0 +1,171 @@
+        awarding_toptier_agency_code=ta1.toptier_code,
+        awarding_toptier_agency_name="Department",
+        funding_agency_id=ag2.id,
+        funding_toptier_agency_code=ta2.toptier_code,
+    )
+    award5 = baker.make(
+        awards_lazy_ref,
+        award_id=5,
+        type="08",
+        date_signed="2020-12-15",
+        action_date="2020-12-15",
+        earliest_transaction_id=50,
+        latest_transaction_id=50,
+        awarding_agency_id=ag1.id,
+        awarding_toptier_agency_code=ta1.toptier_code,
+        awarding_toptier_agency_name="Department",
+        funding_agency_id=ag2.id,
+        funding_toptier_agency_code=ta2.toptier_code,
+    )
+    award6 = baker.make(
+        awards_lazy_ref,
+        award_id=6,
+        type="08",
+        date_signed="2021-07-05",
+        action_date="2021-07-05",
+        earliest_transaction_id=60,
+        latest_transaction_id=60,
+        awarding_agency_id=ag3.id,
+        awarding_toptier_agency_code=ta3.toptier_code,
+        awarding_toptier_agency_name="Department of Commerce",
+        funding_agency_id=ag2.id,
+        funding_toptier_agency_code=ta2.toptier_code,
+    )
+
+    transaction_search_lazy_ref = "search.TransactionSearch"
+    baker.make(transaction_search_lazy_ref, transaction_id=10, award=award1, action_date="2019-10-15")
+    baker.make(transaction_search_lazy_ref, transaction_id=20, award=award2, action_date="2020-12-15")
+    baker.make(transaction_search_lazy_ref, transaction_id=30, award=award3, action_date="2020-01-30")
+    baker.make(transaction_search_lazy_ref, transaction_id=40, award=award4, action_date="2019-09-30")
+    baker.make(transaction_search_lazy_ref, transaction_id=50, award=award5, action_date="2020-12-15")
+    baker.make(transaction_search_lazy_ref, transaction_id=60, award=award6, action_date="2021-07-05")
+
+
+@pytest.mark.django_db
+def test_award_count_success(client, monkeypatch, award_data, helpers, elasticsearch_award_index):
+    setup_elasticsearch_test(monkeypatch, elasticsearch_award_index)
+    helpers.mock_current_fiscal_year(monkeypatch)
+
+    resp = client.get(url.format(filters=""))
+    results = resp.data["results"]
+    assert resp.status_code == status.HTTP_200_OK
+    assert len(results) == 1
+
+
+@pytest.mark.django_db
+def test_award_count_specific_year(client, monkeypatch, award_data, helpers, elasticsearch_award_index):
+    setup_elasticsearch_test(monkeypatch, elasticsearch_award_index)
+
+    resp = client.get(url.format(filters="?fiscal_year=2019"))
+    results = resp.data["results"]
+    assert resp.status_code == status.HTTP_200_OK
+    assert len(results) == 1
+
+    resp = client.get(url.format(filters="?fiscal_year=2021"))
+    results = resp.data["results"]
+    assert resp.status_code == status.HTTP_200_OK
+    assert len(results) == 1
+
+
+@pytest.mark.django_db
+def test_award_count_cfo_agencies_only(client, monkeypatch, award_data, helpers, elasticsearch_award_index):
```

Comment:
I know the test is passing, so it must be working, but I don't follow the logic. 
I don't see a year being passed in as a filter, nor do I see the current fiscal year being patched, which I expect would result in the actual current fiscal year (24) being used used here. The Award associated with the Department of Commerce (a CFO agency) takes place in 2021, so I would expect it not to appear in the results.

What am I missing here? Does it have to do with the patched Fiscal Year lingering around from the first test?

Reply:
Good question. This was an oversight. I agree with you that it should require a patched fiscal year. Let me add in the patch

Reply:
Addressed